skimage/restoration/inpaint.py

```diff
@@ -0,0 +1,134 @@
+from __future__ import print_function, division
+
+import numpy as np
+import skimage
+from scipy import sparse
+from scipy.sparse.linalg import spsolve
+from scipy.ndimage.filters import laplace
+
+
+def inpaint_biharmonic(img, mask, multichannel=False):
+    """Inpaint masked points in image with biharmonic equations.
+
+    Parameters
+    ----------
+    img : nD{+color channel} np.ndarray
+        Input image.
+    mask : nD np.ndarray
+        Array of pixels to be inpainted. Have to be the same size as one
+        of the 'img' channels. Unknown pixels has to be represented with 1, 
+        known pixels - with 0.
+    multichannel : boolean, optional
+        If True, the last `img` dimension is considered as a color channel.
+
+    Returns
+    -------
+    out : nD{+color channel} np.array
+        Input image with masked pixels inpainted.
+
+    Example
+    -------
+    >>> img = np.tile(np.square(np.linspace(0, 1, 5)), (5, 1))
+    >>> mask = np.zeros_like(img)
+    >>> mask[2, 2:] = 1
+    >>> mask[1, 3:] = 1
+    >>> mask[0, 4:] = 1
+    >>> out = inpaint_biharmonic(img, mask)
+
+    References
+    ----------
+    Algorithm is based on:
+    .. [1]  N.S.Hoang, S.B.Damelin, "On surface completion and image inpainting
+            by biharmonic functions: numerical aspects",
+            http://www.ima.umn.edu/~damelin/biharmonic
+    """
+    
+    def _inpaint(img, mask):
```

Comment:
Why is this function nested? Usually nesting is done when it saves many input arguments (by using the arguments from the enclosing scope). In this case, you save only one argument, so you might as well pull the function out.

Additionally, you should give it a more descriptive name, in this case, `_inpaint_single_channel` or `_inpaint_greyscale`.